pkg/controller/petset/pet_set.go

```diff
@@ -348,7 +348,13 @@ func (psc *PetSetController) syncPetSet(ps *apps.PetSet, pets []*api.Pod) (int,
 		case deletePet:
 			err = petManager.Delete(pet)
 		}
-		if err != nil {
+		switch err.(type) {
+		case errUnhealthyPet:
+			// We are not passing this error up, but we don't increment numPets if we encounter it,
+			// since numPets directly translates to petset.status.replicas
+		case nil:
+			continue
```

Comment:
slightly prefer continue since it's more explicit, and we can't end up adding some code at the end of the for loop that we execute by mistake in one case and not the other